code_annotations/extensions/base.py

```diff
@@ -102,15 +114,25 @@ def search(self, file_handle):
         if any(anno in txt for anno in self.config.annotation_tokens):
             fname = clean_abs_path(file_handle.name, self.config.source_path)
 
+            # Iterate on all comments: both multi- and single-line.
             for match in self.comment_regex.finditer(txt):
-                # Should only be one match
-                comment_content = [item for item in match.groups() if item is not None][0]
-                for inner_match in self.query.finditer(comment_content):
-                    # Get the line number by counting newlines + 1 (for the first line).
-                    # Note that this is the line number of the beginning of the comment, not the
-                    # annotation token itself.
-                    line = txt.count('\n', 0, match.start()) + 1
+                # Get the line number by counting newlines + 1 (for the first line).
+                # Note that this is the line number of the beginning of the comment, not the
+                # annotation token itself.
+                line = txt.count('\n', 0, match.start()) + 1
+
+                # Find multi-line comment
+                comment_content = match.groupdict()["multiline_comment"]
+                if not comment_content:
+                    # Find single-line comments
+                    comment_content = match.groupdict()["singleline_comment"]
+                    # Strip leading comment symbols
+                    comment_content = self.singleline_comment_regex.sub(
+                        "\n",
+                        comment_content
+                    )
```

Comment:
I think this would read nicer if it were updated to:
```
comment_content = self._get_clean_comment_content(match)
```
And something like:
```
def _get_clean_comment_content(self, match):
    """ Return cleaned comment content """
    comment_content = match.groupdict()["multiline_comment"]
    if comment_content:
        return comment_content

    comment_content = match.groupdict()["singleline_comment"]
    # Strip leading comment symbols
    return self.singleline_comment_regex.sub(
        "\n",
        comment_content
    ) 
```

Reply:
FYI: The comment above uses `single-line comment token` rather than `comment symbol`.  We should choose one and try to be consistent throughout the file.